core/src/main/java/com/datastax/oss/driver/internal/core/config/cloud/CloudConfigFactory.java

```diff
@@ -228,8 +229,18 @@ protected BufferedReader fetchProxyMetadata(
       connection.setSSLSocketFactory(sslContext.getSocketFactory());
       connection.setRequestMethod("GET");
       connection.setRequestProperty("host", "localhost");
-      return new BufferedReader(
-          new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+      try {
+        return new BufferedReader(
+            new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+      } catch (FileNotFoundException e) {
+        throw new IllegalStateException(
+            "Metadata service request path not found. Please make sure the metadata service url is correct",
+            e);
+      } catch (IOException ioe) {
+        throw new IllegalStateException(
+            "Unable to read data from cloud metadata service. Please make sure your cluster is not parked or terminated",
+            ioe);
+      }
```

Comment:
I'd like for more targeted catch blocks here.  As it is the exceptions indicated here could be thrown by either HttpsURLConneciton.getInputStream(), the InputStreamReader constructor or the BufferedReader constructor.